File: articles/conscience-guilt-shame-and-moral-self-evaluation/c/repair_probability.c
/*
Synthetic repair probability from guilt, shame, empathy, and repair pathway.
Educational scaffold only.

Compile:
  cc c/repair_probability.c -lm -o outputs/tables/repair_probability_c
Run:
  ./outputs/tables/repair_probability_c
*/

#include <math.h>
#include <stdio.h>

double logistic(double x) {
    return 1.0 / (1.0 + exp(-x));
}

int main(void) {
    double guilt = 0.8;
    double shame = 0.4;
    double empathic_acknowledgment = 0.7;
    double repair_pathway = 0.6;
    double agency_capacity = 0.6;
    double defensiveness = 0.2;

    double latent =
        0.50 * guilt -
        0.30 * shame +
        0.35 * empathic_acknowledgment +
        0.30 * repair_pathway +
        0.20 * agency_capacity -
        0.25 * defensiveness;

    double probability = logistic(latent);

    printf("Synthetic repair probability: %.3f\n", probability);
    return 0;
}

File: articles/conscience-guilt-shame-and-moral-self-evaluation/rust/repair_probability.rs
// Synthetic repair probability from guilt, shame, empathy, and repair pathway.
// Educational scaffold only.
//
// Compile:
//   rustc rust/repair_probability.rs -o outputs/tables/repair_probability_rust
// Run:
//   ./outputs/tables/repair_probability_rust

fn logistic(x: f64) -> f64 {
    1.0 / (1.0 + (-x).exp())
}

fn main() {
    let guilt = 0.8_f64;
    let shame = 0.4_f64;
    let empathic_acknowledgment = 0.7_f64;
    let repair_pathway = 0.6_f64;
    let agency_capacity = 0.6_f64;
    let defensiveness = 0.2_f64;

    let latent =
        0.50 * guilt
        - 0.30 * shame
        + 0.35 * empathic_acknowledgment
        + 0.30 * repair_pathway
        + 0.20 * agency_capacity
        - 0.25 * defensiveness;

    let probability = logistic(latent);

    println!("Synthetic repair probability: {:.3}", probability);
}
